src/Fixer/ArrayNotation/YieldFromArrayToYieldsFixer.php

```diff
@@ -31,18 +31,26 @@ public function getDefinition(): FixerDefinitionInterface
     {
         return new FixerDefinition(
             'Yield from array must be unpacked to series of yields.',
-            [new CodeSample('<?php function generate() {
+            [
+                new CodeSample('<?php function generate() {
     yield from [
         1,
         2,
         3,
     ];
 }
-')],
-            'The conversion will make the array in `yield from` changed in arrays of 1 less dimension.'
+'),
+            ],
+            'The conversion will make the array in `yield from` changed in arrays of 1 less dimension.',
+            'The rule is risky in case of `yield from` being used multiple times within single function scope and one relies on the list-alike indices (e.g. `function foo() { yield from ["a"]; yield from ["b"]; }`).'
```

Comment:
What's the context of "one relies"? One of multiple yields mentioned before? Personally I would focus on key consuming, like:

```suggestion
            'The rule is risky in case of `yield from` being used multiple times within single function scope, while using list-alike data sources (e.g. `function foo() { yield from ["a"]; yield from ["b"]; }`). It only matters when consuming such iterator with key-value context, because set of yielded keys may be changed after applying this rule.'
```

Reply:
by "one relies on sth" I meant "someone depends on sth".

I like your descr more, thanks